components/base/wheeled/kinematics.go

```diff
@@ -0,0 +1,110 @@
+package wheeled
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/golang/geo/r3"
+	"github.com/pkg/errors"
+	"go.viam.com/rdk/components/base"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/services/slam"
+	"go.viam.com/rdk/spatialmath"
+)
+
+type kinematicWheeledBase struct {
+	*wheeledBase
+	slam  slam.Service
+	model referenceframe.Model
+}
+
+func WrapWithKinematics(base *wheeledBase, slam slam.Service) (base.KinematicBase, error) {
+	var err error
+	kwb := &kinematicWheeledBase{
+		wheeledBase: base,
+		slam:        slam,
+	}
+	kwb.model, err = kwb.buildModel()
+	if err != nil {
+		return nil, err
+	}
+	return kwb, err
+}
+
+func (kwb *kinematicWheeledBase) ModelFrame() referenceframe.Model {
+	return kwb.model
+}
+
+func (kwb *kinematicWheeledBase) CurrentInputs(ctx context.Context) ([]referenceframe.Input, error) {
+	// TODO(RSDK-2311): complete the implementation
+	return []referenceframe.Input{}, errors.New("not implemented yet")
+}
+
+func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, goal []referenceframe.Input) error {
+	// TODO(RSDK-2311): complete the implementation
+	return errors.New("not implemented yet")
+}
+
+func (kwb *kinematicWheeledBase) buildModel() (referenceframe.Model, error) {
+	if kwb.collisionGeometry == nil {
+		return nil, errors.New("cannot create model for base with no collision geometry")
+	}
+
+	// get the limits of the SLAM map to set as the extents of the frame
+	data, err := slam.GetPointCloudMapFull(context.Background(), kwb.slam, "")
```

Comment:
Oops.  I should probably change this so that it passes the context along from where it is ultimately instantiated, huh?

Reply:
My thought was to give it motion planning's context and have it be entirely separate from SLAM, but since we each have separate ideas of what to do here maybe this warrants deeper thought.